### PR TITLE
Restrict posting actions to email threads

### PR DESCRIPTION
**Pull request by ivankuznetsov**
## Summary

`hey move`, `hey trash`, and `hey ignore` now require the exact flag `--kind topic`. Missing and non-email kinds fail before config, authentication, or any SDK request, while existing bulk-ID behavior stays unchanged.

Fresh `main` already contains the posting actions from #173, #175, and #176. This rebase drops PR #152's superseded command implementations and applies only the email-thread safety guard to the current commands. The CLI does not manage HEY World posts.

## Validation

- `make fmt-check`
- `make vet`
- `go test ./internal/...`
- `make tidy-check`
- `make check-surface`
- `go test -race -count=1 ./internal/...`
- `make build`
- A malformed-config regression proves kind validation runs before root setup
- Compound Engineering review completed; its `id` versus `topic_id` wording finding was fixed and regression-tested

## Risk

`--kind` is a caller-supplied safety assertion, not an authorization boundary. The server remains responsible for rejecting an ID that is invalid for an action.

---
## Summary by cubic
Restricts `move`, `trash`, and `ignore` to email threads and fails fast on non-email kinds. Previously these commands ran with any box item ID; now they require `--kind topic` and reject `world/post` and other kinds before config, auth, or any request.

- Callers must pass `--kind topic` with box item IDs (`id`) for `hey move`, `hey trash`, and `hey ignore`. `spam` and `stop-ignoring` are unchanged.
- Adds `emailPostingArgs` to validate the kind early and return clearer errors, including a specific message for HEY World posts.
- Updates CLI help, README, `.surface`, breadcrumbs, and examples to show `--kind topic`; help notes clarify using `id` vs `topic_id` and how to source `id` from box/label/search JSON.
- Adds unit tests verifying early kind validation (even with malformed config) and updated help content.

<sup>Written for commit 22a946a1605706b61d94d5b0fa5b6e7c4f6d9100. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/basecamp/hey-cli/pull/152?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: internal/cmd/posting_actions.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+	return writeOK(nil, output.WithSummary(summary))
+}
+
+func newPaperTrailCommand() *postingActionCommand {
+	return newPostingActionCommand(
+		"paper-trail",
+		"Move postings to Paper Trail",
+		"moved to Paper Trail",
+		"State-changing action. Confirm the exact posting IDs and obtain explicit user approval before moving each posting to Paper Trail.",
```
[the remark this answers is not in the document]
Comment (ivankuznetsov) [reply internal/cmd/posting_actions.go]: > make both descriptions singular

Addressed in 9b5e7f8 — all six command descriptions and machine-readable `agent_notes` now use singular posting/ID wording, with regression coverage for the metadata.

File: internal/cmd/posting_actions.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+	}
+
+	summary := fmt.Sprintf("Posting %d %s", ids[0], c.pastTense)
+
+	if writer.IsStyled() {
+		fmt.Fprintln(cmd.OutOrStdout(), summary+".")
+		return nil
+	}
+
+	return writeOK(nil, output.WithSummary(summary))
```
[the remark this answers is not in the document]
Comment (ivankuznetsov) [reply internal/cmd/posting_actions.go:66]: > Validate/reject these formats before the SDK call, or return list-shaped action data that they can render.

Addressed in ef215dc: the shared posting-action runner now rejects `--ids-only` and `--count` before calling the SDK. The regression test asserts that neither format sends a server request.

**Comment by code-monger-givenall** [comment]
I ran into one edge case while testing this against a mixed Imbox. `hey box` can return `world/post` records alongside email topics. Passing one of those numeric IDs to `hey trash` gets a successful response from the email trash endpoint, leaves the World post untouched, and currently prints that it moved to Trash.

I opened [ivankuznetsov/hey-cli#1](https://github.com/ivankuznetsov/hey-cli/pull/1) against this branch. It requires callers to pass the posting kind and rejects `world/post` before authentication or any HTTP request. The full check and race suites pass, and a live guard check logged no request. No World content was changed.

**Comment by jeremy** [comment]
Thanks for the rebase work here, and for keeping this alive after the posting actions themselves landed through #173, #175 and #176. Having read what remains against current `main`, I'm going to decline this one, and I want to be specific about why so it isn't re-litigated.

What's left in the PR is a mandatory `--kind topic` on `hey move`, `hey trash` and `hey ignore`. That breaks every existing caller of three commands in exchange for a guard the CLI cannot check: the flag is a promise the caller makes about the ID, and the PR's own Risk section says as much. A caller who passes a HEY World box item today gets a misleading success; with this change a caller who passes the same ID plus `--kind topic` gets the same misleading success. The failure it stops is the one where a caller already knows the kind and types it in, which is the caller who least needs stopping.

The underlying case is real. Box postings do carry `kind`, `world/post` rows do appear alongside email in a box, and the bulk trash and spam endpoints answer 2xx for one without acting on it. The place to close that is the listing (show the kind so a reader can tell) and the server (reject or handle a World posting on an email action), not a required flag on the verbs. #185 carries the listing and TUI side of that, and the verdict there narrows it accordingly.

The branch is also conflicting with `main`, which has since moved the listing and posting code around (`hey box` became `hey box view`, listings live in `internal/cmd/postings_listing.go` and `internal/mail`). If you'd like to keep contributing in this area, the restore and system-view PRs from #186 and #161 are the ones with a clear path.
